### PR TITLE
feat(ESSNTL-4054): Add modified_on timestamp filter

### PR DESCRIPTION
**Pull request by kruai**
Addresses [ESSNTL-4054](https://issues.redhat.com/browse/ESSNTL-4054).
Add a timestamp filter for the host's `modified_on` field. Also adds some tests to validate gte, lte, and gt + lt functions.

### CODE REVIEW

File: src/resolvers/hosts/hosts.integration.ts [hunk 1]
```diff
@@ -387,6 +388,39 @@ describe('hosts query', function () {
...
+            test('gte', async () => {
+                const { data } = await runQuery(BASIC_QUERY,
+                    { filter: { modified_on: {
+                        gte: '2020-03-10T08:07:00.000000Z'
+                    }}}
+                );
+                data.hosts.data.should.have.length(1);
+                data.hosts.data[0].id.should.equal(TEST_ORG_ID_HOST_IDS[4]);
+            });
+
```
Comment (CodeHeeler) [inline src/resolvers/hosts/hosts.integration.ts:411]: I think you would also want to test the strictly great and lesser than in addition to equals for the gte and lte cases as well as some negative equivalents to "assertNotEqual" to be sure it works as expected for all cases, not just one for each operator.

**Review by CodeHeeler — APPROVED** [review]
Overall looks good.  I even checked each test host id and timestamp against your logic to make sure the right ones were selected.

Left one question regarding adding additional test cases to cover every possibility, but this is non-blocking.

File: src/resolvers/hosts/hosts.integration.ts (hunk 1 above)
Comment (kruai) [reply src/resolvers/hosts/hosts.integration.ts:411]: @CodeHeeler `modified_on` is using the same `FilterTimestamp` type that's already tested a few times, so I wasn't sure if it was necessary to test it beyond the most common use-cases. I can add a couple more tests if you think it's worth it though

Comment (kruai) [reply src/resolvers/hosts/hosts.integration.ts:411]: What do you mean by negative equivalents in this case, though? Each of them is looking for one specific seeded host. If only one host is returned, that to me seems like it covers the negative as well (all of the other hosts were _not_ returned). wdyt?

Comment (CodeHeeler) [reply src/resolvers/hosts/hosts.integration.ts:411]: I'm ok with leaving this as is then.  Thanks for the explanation.

Comment (CodeHeeler) [reply src/resolvers/hosts/hosts.integration.ts:411]: Ah, yes, you're right, we're testing the negatives as well by the omission of the other hosts, that's true.

Comment (kruai) [reply src/resolvers/hosts/hosts.integration.ts:431]: I added separate LT & GT tests just to make sure we're covered.

Comment (CodeHeeler) [reply src/resolvers/hosts/hosts.integration.ts:431]: Always better to have more test coverage than too little.  Thank you for this.

**Review by FabriciaDinizRH — APPROVED** [review]
LGTM
